### DISCUSSION TITLE
Insert Tags at cursor position from textarea and or markdown editor

### DISCUSSION THREAD

**Discussion by momostafa** [filamentphp/filament#13300] [Ideas]
Hello,

Message and email personalization is already essential part of any application that interacts with users, teams etc...
it would be great to have native functionality to insert Tags at cursor point when using textarea or any available text editor like example screen recording below.

Thanks in advance for your support and considerations.

https://github.com/filamentphp/filament/assets/12662539/bccc18e6-bb4a-45b4-9465-53a9daece26c

**Comment by leandrocfe** [comment]
I believe this might be a plugin rather than a native functionality.

**Reply by momostafa** [reply]
I hope one of the developers who are fluent with Livewire and AlpineJs will bring this idea to live... meanwhile I have already started learning about Livewire which sounds very interesting but i am still not sure about Alpine as I am not a JS guy in general  and hopefully implement it myself.

As of now my workaround is as below

```php
                    Forms\Components\Select::make('sms_template_id')
                    ->live(onBlur: true)
                    ->label('SMS Templates')
                    ->options(fn (Get $get) => SmsTemplate::where('user_id', $get('user_id'))
                    ->where('active', 1)
                    ->pluck('template_name','id'))
                    ->afterStateUpdated(
                        function (Set $set, string $state){
                        $template = SmsTemplate::where('id', $state)->first();
                        $set('sms_message', $template->sms_message); // example Dear %FIRST_NAME% , Your OTP is %UNIQUE_OTP%
                        }
                    )
```
